ospd_openvas/openvas_db.py

```diff
@@ -60,219 +44,234 @@
-    """
-    ctx = get_kb_context()
-    return ctx.srandmember(name)
-
-def item_add_single(name, values):
-    """ Add a single KB element with one or more values.
-    The right global REDISCONTEXT must be already set.
-    """
-    ctx = get_kb_context()
-    ctx.sadd(name, *set(values))
-
-def item_set_single(name, value):
-    """ Set (replace) a new single KB element. The right global
-    REDISCONTEXT must be already set.
-    """
-    ctx = get_kb_context()
-    pipe = ctx.pipeline()
-    pipe.delete(name)
-    pipe.sadd(name, *set(value))
-    pipe.execute()
-
-def item_del_single(name):
-    """ Delete a single KB element. The right global REDISCONTEXT must be
-    already set.
-    """
-    ctx = get_kb_context()
-    ctx.delete(name)
-
-def get_pattern(pattern):
-    """ Get all items stored under a given pattern.
-    """
-    ctx = get_kb_context()
-    items = ctx.keys(pattern)
-
-    elem_list = []
-    for item in items:
-        elem_list.append([item, ctx.smembers(item)])
-    return elem_list
-
-def get_elem_pattern_by_index(pattern, index=1):
-    """ Get all items with index 'index', stored under
-    a given pattern.
-    """
-    ctx = get_kb_context()
-    items = ctx.keys(pattern)
-
-    elem_list = []
-    for item in items:
-        elem_list.append([item, ctx.lindex(item, index)])
-    return elem_list
-
-def release_db(kbindex=0):
-    """ Connect to redis and select the db by index.
-    Flush db and delete the index from DBINDEX_NAME list.
-    """
-    if kbindex:
-        ctx = kb_connect(kbindex)
-        ctx.flushdb()
-        ctx = kb_connect()
-        ctx.hdel(DBINDEX_NAME, kbindex)
-
-def get_result():
-    """ Get and remove the oldest result from the list. """
-    ctx = get_kb_context()
-    return ctx.rpop("internal/results")
-
-def get_status():
-    """ Get and remove the oldest host scan status from the list. """
-    ctx = get_kb_context()
-    return ctx.rpop("internal/status")
-
-def get_host_scan_scan_start_time():
-    """ Get the timestamp of the scan start from redis. """
-    ctx = get_kb_context()
-    return ctx.rpop("internal/start_time")
-
-def get_host_scan_scan_end_time():
-    """ Get the timestamp of the scan end from redis. """
-    ctx = get_kb_context()
-    return ctx.rpop("internal/end_time")
+    def kb_new(self):
+        """ Return a new kb context to an empty kb.
+        """
+        ctx = self.db_find(self.dbindex_name)
+        for index in range(1, self.max_dbindex):
+                if self.try_database_index(ctx, index) == 1:
+                    ctx = self.kb_connect(index)
+                    return ctx
+
+    def get_kb_context(self):
+        """ Get redis context if it is already connected or do a connection.
+        """
+        if self.rediscontext is not None:
+            return self.rediscontext
+
+        self.rediscontext = self.db_find(self.dbindex_name)
+
+        if self.rediscontext is None:
+            print("Problem retrieving Redis Context")
```

Comment:
Should be a logging statement instead of print. A user may not have a console or the error should be logged to a file.